### PR TITLE
chore: update webgpu

### LINKED ISSUE #10098: WebGPU: Attempting to use mapped buffer with GPU panics
Some meta:
- MacBook Pro (16-inch, 2019) macOS Big Sur
- Intel UHD Graphics 630 1536 MB

```console
$ deno --version
deno 1.8.3 (release, x86_64-apple-darwin)
v8 9.0.257.3
typescript 4.2.2
```

Seeing the following panic:

```txt
thread 'main' panicked at 'Buffer Valid((3, 1, Metal)) is still mapped', /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-core-0.7.1/src/device/queue.rs:548:42
stack backtrace:
   0: _rust_begin_unwind
   1: std::panicking::begin_panic_fmt
   2: wgpu_core::device::queue::<impl wgpu_core::hub::Global<G>>::queue_submit
   3: deno_core::ops_json::json_op_sync::{{closure}}
   4: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
   5: deno_runtime::metrics::metrics_op::{{closure}}
   6: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
   7: <extern "C" fn(A0) .> R as rusty_v8::support::CFnFrom<F>>::mapping::c_fn
   8: __ZN2v88internal25FunctionCallbackArguments4CallENS0_15CallHandlerInfoE
   9: __ZN2v88internal12_GLOBAL__N_119HandleApiCallHelperILb0EEENS0_11MaybeHandleINS0_6ObjectEEEPNS0_7IsolateENS0_6HandleINS0_10HeapObjectEEESA_NS8_INS0_20FunctionTemplateInfoEEENS8_IS4_EENS0_16BuiltinArgumentsE
  10: __ZN2v88internalL26Builtin_Impl_HandleApiCallENS0_16BuiltinArgumentsEPNS0_7IsolateE
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
fatal runtime error: failed to initiate panic, error 5
```

After failing to `unmap` a storage copy source buffer and then attempting to use it's contents with the GPU. I will endeavour to add a small reproduceable example to the issue when it isn't quite so late!

### LINKED ISSUE #12721: bug(ext/webgpu): requiredFeatures don't work
Even after specifying a required feature in `GPUDeviceDescriptor`, the device's feature field contains no feature. Not only that, in my case, timestamp query related things seem to fail silently (but can be patched to throw the errors).

#### Repro 1, `device.features` don't contain the required feature

```ts
const adapter = (await navigator.gpu.requestAdapter())!;
console.log("adapter has timestamp query feature:", adapter.features.has("timestamp-query"));

const device = (await adapter.requestDevice({
  requiredFeatures: ["timestamp-query"],
}))!;
console.log("device has timestamp query feature:", device.features.includes("timestamp-query"));
```

Output:

image

#### Repro 2, `device.createQuerySet` silently fails in validation

```ts
import { enableValidationErrors } from "https://crux.land/gpu_err@1.0.0";

const adapter = (await navigator.gpu.requestAdapter())!;

const device = (await adapter.requestDevice({
  requiredFeatures: ["timestamp-query"],
}))!;

enableValidationErrors(device, true);

const querySet = device.createQuerySet({
  type: "timestamp",
  count: 1,
});
```

Output:

image

deno: v1.16.0
os: windows
webgpu backend: Vulkan, gpu: Intel

### LINKED ISSUE #17580: Update wgpu for TersorflowJS WebGPU support
In mid 2022, I wanted to migrate my tensorflow.js projects into Deno, only to find that their WebGPU backend is mysteriously failing https://github.com/tensorflow/tfjs/issues/6842.

It turns out `naga` was interpreting wgsl in lexical order https://github.com/gfx-rs/naga/pull/2075, luckily the new module level scoping has been released in `naga` 0.11 and `wgpu` 0.15.

Is it possible for Deno to update `wgpu` to 0.15?

### PR DESCRIPTION
**Pull request by crowlKats**
Fixes #10098
Fixes #12721
Closes #17580

### CODE REVIEW

File: ext/webgpu/Cargo.toml [hunk 1]
```diff
@@ -3,16 +3,38 @@
...
+
 [dependencies]
 deno_core.workspace = true
-serde.workspace = true
-tokio.workspace = true
-wgpu-core = { version = "0.13", features = ["trace", "replay", "serde"] }
-wgpu-types = { version = "0.13", features = ["trace", "replay", "serde"] }
+raw-window-handle = { workspace = true, optional = true }
+serde = { workspace = true, features = ["derive"] }
+tokio = { workspace = true, features = ["full"] }
```
Comment (bartlomieju) [inline ext/webgpu/Cargo.toml:20]: Why need to add these features here?

File: ext/webgpu/src/lib.rs [hunk 2]
```diff
@@ -65,6 +76,8 @@ pub mod queue;
 pub mod render_pass;
 pub mod sampler;
 pub mod shader;
+#[cfg(feature = "surface")]
```
Comment (bartlomieju) [inline ext/webgpu/src/lib.rs:79]: Why is it hidden behind a feature flag? Is this flag on by default?

**Review by bartlomieju — COMMENTED** [review]
What's the size of resulting binary produced by this change?

File: ext/webgpu/src/lib.rs (hunk 2 above)
Comment (crowlKats) [reply ext/webgpu/src/lib.rs:79]: no its disabled by default. its not important directly to cli, but for other projects

File: ext/webgpu/Cargo.toml (hunk 1 above)
Comment (crowlKats) [reply ext/webgpu/Cargo.toml]: fixed

**Comment by crowlKats** [comment]
> What's the size of resulting binary produced by this change?

seems to be about 0.3 mb less than main

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**